source/internal.d.ts

```diff
@@ -328,3 +408,283 @@ IsPrimitive<Object>
 ```
 */
 export type IsPrimitive<T> = [T] extends [Primitive] ? true : false;
+
+/**
+Returns a boolean for whether A and B are both true.
+
+@example
+```
+And<true, true>
+//=> true
+
+And<true, false>
+//=> false
+```
+*/
+export type And<A extends boolean, B extends boolean> = [A, B][number] extends true
+	? true
+	: true extends [IsEqual<A, false>, IsEqual<B, false>][number]
+		? false
+		: boolean;
+
+/**
+Returns a boolean for either A or B is true.
+
+@example
+```
+Or<true, false>
+//=> true
+
+Or<false, false>
+//=> false
+```
+*/
+export type Or<A extends boolean, B extends boolean> = [A, B][number] extends false
+	? false
+	: true extends [IsEqual<A, true>, IsEqual<B, true>][number]
+		? true
+		: boolean;
+
+/**
+Returns a boolean for whether A is false.
+
+@example
+```
+Not<true>
+//=> false
+
+Not<false>
+//=> true
+```
+*/
+export type Not<A extends boolean> = A extends true
+	? false
+	: A extends false
+		? true
+		: boolean;
+
+/**
+Returns the maximum value from a tuple of integers.
+
+Note:
+- float number is not supported.
+
+@example
+```
+ArrayMax<[1, 2, 5, 3]>;
+//=> 5
+
+ArrayMax<[1, 2, 5, 3, 99, -1]>
+//=> 99
+```
+*/
+export type ArrayMax<A extends number[], Result extends number = NegativeInfinity> = A extends [infer F extends number, ...infer R extends number[]]
+	? Gt<F, Result> extends true
+		? ArrayMax<R, F>
+		: ArrayMax<R, Result>
+	: Result;
+
+/**
+Returns the minimal value from a tuple of integers.
+
+@example
+```
+ArrayMin<[1, 2, 5, 3]>
+//=> 1
+
+ArrayMin<[1, 2, 5, 3, -5]>
+//=> -5
+```
+*/
+export type ArrayMin<A extends number[], Result extends number = PositiveInfinity> = A extends [infer F extends number, ...infer R extends number[]]
+	? Lt<F, Result> extends true
+		? ArrayMin<R, F>
+		: ArrayMin<R, Result>
+	: Result;
+
+/**
+Returns the absolute value of a given value.
+
+@example
+```
+NumberAbsolute<-1>
+//=> 1
+
+NumberAbsolute<1>
+//=> 1
+
+NumberAbsolute<NegativeInfinity>
+//=> PositiveInfinity
+```
+*/
+export type NumberAbsolute<N extends number> = `${N}` extends `-${infer StringPositiveN}` ? StringToNumber<StringPositiveN> : N;
+
+/**
+Returns a boolean for whether A > B.
+
```

Comment:
Specifying the range would be better

Reply:
done

Reply:
it already supports large integer, I removed the todo comment.